kernel/include/kernel/thread.h

```diff
@@ -106,6 +106,10 @@ typedef struct thread {
     /* thread local storage */
     uintptr_t tls[MAX_TLS_ENTRY];
 
+    /* Flag to indicate that interrupt
+     * context is active for the thread */
+    uint32_t intr_ctxt_active;
+
     char name[32];
```

Comment:
Probably this could have been better handled by per CPU interrupt state.

Reply:
Also problematic is it is only implemented on ARM. For a new feature like this it really should be implemented on all of the architectures, at least if possible.

IIRC, there's an arch_in_interrupt_handler() routine that may be actually what you're looking for. Or at least having it be an arch_ routine that can be stubbed out by default if the arch doesn't want to implement it. I'm leery about adding state at the thread level.

Reply:
Hmm, actually looks like the arch_in_int_handler() is only basically implemented on cortex-m, since the 'big arm' one implements it as a global bool which is clearly not SMP safe. Looks like might make sense to fix that first.

Reply:
Thanks Travis for the feedback, makes sense. I will try to upload something for all the archs, assuming no arch wants to allow context switching to happen between IRQ handling.